File: Python/grade-calculator.py
## GRADE CALCULATOR ##
## THIS PROGRAM CALCULATES YOUR AVERAGE PER SUB FOR THE WHOLE SEM ##
## AVERAGE FOR ONE SEM IS ALSO COMPUTED ##

print("\tGRADE CALCULATOR\n")

grades = []
terms = []

n = int (input("ENTER NO. OF SUBJECTS: "))

for i in range (0, n):
    print ("\nSubject")
    for j in range (3):
        term = float (input("Grade: "))
        terms.append(term)
    grade = sum(terms) / len(terms) ##avg
    grades.append(grade)
    terms.clear()

    print ("GRADE: " + "{:.2f}".format(grade))
   ## print (terms)

gen_ave = sum(grades) / len (grades)
print("\nGEN AVERAGE: " + "{:.2f}".format(gen_ave))

File: Java/GradeCalculator.java
import java.util.*;
public class GradeCalculator {
    public static void main(String[] args) {
        /*THIS IS A VERY SIMPLE GRADE CALCULATOR
        THAT UTILIZES ARRAYS*/

        //INITIALIZE VARIABLES
        int n = 10;
        int [] gradeArr = new int[n]; 
        int lowest = gradeArr[0]; int highest = gradeArr[0];
        Scanner input = new Scanner (System.in);
        int average = 0; int sum = 0;

        //Input
        for (int i = 0; i < n; i++)
        {
            System.out.print("Enter your grade: ");
            gradeArr[i] = input.nextInt();
        }
        
        //Average
        for (int i = 0; i < n; i++)
        {
            sum = sum + gradeArr[i];
            average = sum / n;
        }
        
        //Lowest, Highest
        for (int i = 0; i < n; i++)
        {
            if (n > highest)
                highest = gradeArr[i];
            else
                lowest = gradeArr[i];
        }

        //Output
        System.out.println("Average: " + average);
        System.out.println("Highest Grade: " + highest);
        System.out.println("Lowest Grade: " + lowest);

    }
}
